src/validation/__tests__/ScalarLeafsRule-test.ts

```diff
@@ -35,6 +42,35 @@ describe('Validate: Scalar leafs', () => {
     ]);
   });
 
+  it('object type having only one selection', () => {
+    const doc: DocumentNode = {
+      kind: Kind.DOCUMENT,
+      definitions: [
+        {
+          kind: Kind.OPERATION_DEFINITION,
+          operation: OperationTypeNode.QUERY,
+          selectionSet: {
+            kind: Kind.SELECTION_SET,
+            selections: [
+              {
+                kind: Kind.FIELD,
+                name: { kind: Kind.NAME, value: 'human' },
+                selectionSet: { kind: Kind.SELECTION_SET, selections: [] },
+              },
+            ],
+          },
+        },
+      ],
+    };
+    const errors = validate(testSchema, doc, [ScalarLeafsRule]);
```

Comment:
maybe just add a comment here that we can't use expectValid like most of the rules do because the parser would throw an error?

does this situation come up anywhere else?

Reply:
I haven't noticed it elsewhere, added the comment!